symbolic-ppdb/src/format/mod.rs

```diff
@@ -276,6 +284,22 @@ impl<'data> PortablePdb<'data> {
                 _ => return Err(FormatErrorKind::UnknownStream.into()),
             }
         }
+
+        // Read source link mappings.
+        // https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers
+        const SOURCE_LINK_KIND: Uuid = uuid::uuid!("CC110556-A091-4D38-9FEC-25AB9A351A6A");
+        let mut source_link_mappings = Vec::new();
+        for cdi in CustomDebugInformationIterator::new(&result, SOURCE_LINK_KIND)? {
+            let cdi = cdi?;
+            // Note: only handle module #1 (do we actually handle multiple modules in any way??)
+            if let (metadata::CustomDebugInformationTag::Module, 1) = (cdi.tag, cdi.value) {
+                let json = String::from_utf8_lossy(result.get_blob(cdi.blob)?);
+                source_link_mappings.push(json);
+            }
+        }
+        result.source_link_mappings =
+            SourceLinkMappings::new(source_link_mappings.iter().map(|v| v.as_ref()))?;
```

Comment:
> IMO doing a [docs.rs/serde_json/latest/serde_json/fn.from_slice.html](https://docs.rs/serde_json/latest/serde_json/fn.from_slice.html) here might be clearer than using from_utf8_lossy (we expect this to be valid utf8 / JSON and might as well error out if its not).

I've originally tried that but abandoned it because I'm transforming that to a custom format anyway.	

> you might as well just take a Vec<Value>.

That would require all the callers to use serde to parse the json. Current solution is simple, just pass the json itself.

> Or alternatively a Vec<&[u8]> if you do want to do the parsing inside of SourceLinkMappings.

OK, changed.

> Also, can we even have more than one mapping in one file?

Not sure, though [this implies there could be](https://github.com/dotnet/sourcelink/pull/605#issuecomment-619402118). In any case, it's not much more complex to handle multiple files so I'd keep the logic in.